lonestar/libdistbench/include/DistBenchStart.h

```diff
@@ -227,27 +213,28 @@ template <typename NodeData, typename EdgeData, bool iterateOutEdges = true>
 std::pair<galois::graphs::DistGraph<NodeData, EdgeData>*,
           galois::graphs::GluonSubstrate<
               galois::graphs::DistGraph<NodeData, EdgeData>>*>
-distGraphInitialization(struct CUDA_Context** GALOIS_USED_ONLY_IN_HETEROGENEOUS(
-                            cuda_ctx) = nullptr) {
+#ifdef GALOIS_ENABLE_GPU
+distGraphInitialization(struct CUDA_Context** cuda_ctx) {
+#else
+distGraphInitialization() {
+#endif
   using Graph     = galois::graphs::DistGraph<NodeData, EdgeData>;
   using Substrate = galois::graphs::GluonSubstrate<Graph>;
   std::vector<unsigned> scaleFactor;
   Graph* g;
   Substrate* s;
 
-#ifdef __GALOIS_HET_CUDA__
+#ifdef GALOIS_ENABLE_GPU
   internal::heteroSetup(scaleFactor);
-  g = loadDGraph<NodeData, EdgeData, iterateOutEdges>(scaleFactor, cuda_ctx);
-#else
-  g = loadDGraph<NodeData, EdgeData, iterateOutEdges>(scaleFactor);
 #endif
+  g = loadDGraph<NodeData, EdgeData, iterateOutEdges>(scaleFactor);
```

Comment:
Correct me if wrong: cuda_ctx now passed in via distGraphInit instead of loadDGraph. Much cleaner. 👍

Reply:
Yes. Loading from file is not related to GPU.